modules/viz/include/opencv2/viz/types.hpp

```diff
@@ -119,6 +119,12 @@ namespace cv
         class CV_EXPORTS Mesh
         {
         public:
+            enum FileTypeReader
+            {
+                PLY,
+                OBJ
+            };
```

Comment:
If the problem is the load method for the binary comparability, I can leave the method with the old interface ```Mesh load(const String& file)`` and I can add into the method an extension parser to get the file type.
On vtk5.8 have you some ideas as workaround?